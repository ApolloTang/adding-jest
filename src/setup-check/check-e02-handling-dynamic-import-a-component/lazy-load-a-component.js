import React from 'react';
import ReactDom from 'react-dom';
import Loadable from 'react-loadable';

const promise = new Promise((rs, rj)=>{
  const content = import('./component-to-lazy-load.js')
  setTimeout(()=>{ rs(content) }, 2000)
});

const MyComponent = Loadable({
  loader: ()=> promise,
  loading: ()=> <div><p>...loading</p></div>
});

export default MyComponent;
export {promise}

