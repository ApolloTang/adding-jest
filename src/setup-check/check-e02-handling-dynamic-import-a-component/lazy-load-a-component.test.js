import React from 'react';
import ReactDom from 'react-dom';

import 'react-testing-library/cleanup-after-each';
import {render} from 'react-testing-library';

import LazyLoadAComponent from './lazy-load-a-component.js';
import {promise} from './lazy-load-a-component.js';

test('lazy load a component', async done => {

  const {container:beforeLoad} = render(<LazyLoadAComponent />);
  expect(beforeLoad.firstChild).toMatchSnapshot();

  await promise;

  const {container:afterLoad} = render(<LazyLoadAComponent />);
  expect(afterLoad.firstChild).toMatchSnapshot();

  done();
})



