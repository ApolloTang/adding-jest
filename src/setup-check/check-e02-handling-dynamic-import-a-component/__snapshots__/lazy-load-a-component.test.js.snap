// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy load a component 1`] = `
<div>
  <p>
    ...loading
  </p>
</div>
`;

exports[`lazy load a component 2`] = `
<div>
  lazy loaded content
</div>
`;
